llvm/lib/IR/Verifier.cpp

```diff
@@ -4392,6 +4392,11 @@ void Verifier::visitAllocaInst(AllocaInst &AI) {
     verifySwiftErrorValue(&AI);
   }
 
+  if (TT.isAMDGPU()) {
+    Check(AI.getAddressSpace() == DL.getAllocaAddrSpace(),
```

Comment:
If we support the 0 case, we should still probably try to filter the unreasonable address spaces. only 0 and 5 would be legitimate

Reply:
I will update this PR once https://github.com/llvm/llvm-project/pull/136584 is merged.